src/perception/semantic_segmentation/semantic_segmentation/segmentation_node.py

```diff
@@ -0,0 +1,142 @@
+from sensor_msgs.msg import Image, CompressedImage
+from std_msgs.msg import String
+from cv_bridge import CvBridge
+from rich.logging import RichHandler
+import logging
+from ament_index_python.packages import get_package_share_directory
+# Configure the root logger
+logging.basicConfig(level=logging.INFO, handlers=[RichHandler(level=logging.WARNING)])
+
+
+# ALGO_VERSION = os.getenv("MODEL_NAME")
+
+# if not ALGO_VERSION:
+#     ALGO_VERSION = 'nvidia/segformer-b2-finetuned-ade-512-512'
+
+package_name = 'semantic_segmentation'
+package_share_directory = get_package_share_directory(package_name)
+CONFIG = os.path.join(package_share_directory, 'resource', 'model',
+                      'segformer_mit-b2_8xb1-160k_cityscapes-1024x1024.py')
+CHECKPOINT = os.path.join(package_share_directory, 'resource', 'model',
+                          'segformer_mit-b2_8x1_1024x1024_160k_cityscapes_20211207_134205-6096669a.pth')
+IMAGE_H = 900
+IMAGE_W = 1600
+
+# Adjust logging
+logging.getLogger('mmseg.apis').setLevel(logging.WARNING)
+logging.getLogger('mmengine').setLevel(logging.WARNING)
+
+"""Cityscapes palette that maps each class to RGB values."""
+COLOR_PALLETE = [
+    [128, 64, 128],  # road
+    [244, 35, 232],  # sidewalk
+    [70, 70, 70],    # building
+    [102, 102, 156],  # wall
+    [190, 153, 153],  # fence
+    [153, 153, 153],  # pole
+    [250, 170, 30],  # traffic light
+    [220, 220, 0],   # traffic sign
+    [107, 142, 35],  # vegetation
+    [152, 251, 152],  # terrain
+    [0, 130, 180],   # sky
+    [220, 20, 60],   # person
+    [255, 0, 0],     # rider
+    [0, 0, 142],     # car
+    [0, 0, 70],      # truck
+    [0, 60, 100],    # bus
+    [0, 80, 100],    # train
+    [0, 0, 230],     # motorcycle
+    [119, 11, 32],   # bicycle
+    [0, 0, 0]        # ignore/unlabeled
+]
+
+
+class SemanticSegmentation(Node):
+    def __init__(self):
+        super().__init__('semantic_segmentation_node')
+        self.declare_parameter('pub_image', True)
+        self.declare_parameter('pub_masks', True)
+        self.declare_parameter('compressed', True)
+        self.declare_parameter('config', "model/segformer_mit-b2_8xb1-160k_cityscapes-1024x1024.py")
+        self.declare_parameter(
+            'checkpoint', "model/segformer_mit-b2_8x1_1024x1024_160k_cityscapes_20211207_134205-6096669a.pth")
+
+        self.compressed = self.get_parameter('compressed').value
+        self.image_subscription = self.create_subscription(
+            Image if not self.compressed else CompressedImage,
+            "/CAM_FRONT/image_rect_compressed",
+            self.listener_callback,
+            qos_profile=QoSProfile(
+                reliability=QoSReliabilityPolicy.RELIABLE,
+                history=QoSHistoryPolicy.KEEP_LAST,
+                depth=0,
+            ),
+        )
+
+        self.image_publisher = self.create_publisher(
+            Image,
+            '/camera/left/segmentations',
+            10
+        )
```

Comment:
instead of outputting a segmented image, I would instead store the masks, in a custom message